tools/src/main/scala/scala/scalanative/build/Build.scala

```diff
@@ -53,60 +58,87 @@ object Build {
    *  @return
    *    [[Config#artifactPath]], the path to the resulting native binary.
    */
-  def build(config: Config)(implicit scope: Scope): Path =
+  def build(config: Config)(implicit scope: Scope): Path = {
+    def makeKey(config: Config): Int = {
+      // skip the whole nativeLink process if the followings are unchanged since the previous build
+      // - build configuration
+      // - class paths' mtime
+      // - the output native binary ('s mtime)
+      // Since the NIR code is shipped in jars, we should be able to detect the changes in NIRs.
+      // One thing we miss is, we cannot detect changes in c libraries somewhere in `/usr/lib`.
+      val key = (
+        config.toString, // we use toString because `config` object contains Logger instance that changes for every invocation
```

Comment:
We can make `logger` an internal field of `Config`, which would not be a field of the constructor. Any interaction with it should happen using `withLogger` accessor. That should simplify and possibly save a few cycles of the CPU.